cinema/tests/test_movie_viewset.py

```diff
@@ -0,0 +1,256 @@
+from cinema.serializers import MovieDetailSerializer, MovieListSerializer
+
+MOVIE_URL = reverse("cinema:movie-list")
+MOVIE_SESSION_URL = reverse("cinema:moviesession-list")
+
+
+def sample_movie(**params):
+    defaults = {
+        "title": "Sample movie",
+        "description": "Sample description",
+        "duration": 90,
+    }
+    defaults.update(params)
+    return Movie.objects.create(**defaults)
+
+
+def sample_genre(**params):
+    defaults = {
+        "name": "Drama",
+    }
+    defaults.update(params)
+    return Genre.objects.create(**defaults)
+
+
+def sample_actor(**params):
+    defaults = {"first_name": "George", "last_name": "Clooney"}
+    defaults.update(params)
+    return Actor.objects.create(**defaults)
+
+
+def sample_movie_session(**params):
+    cinema_hall = CinemaHall.objects.create(
+        name="Blue", rows=20, seats_in_row=20
+    )
+    defaults = {
+        "show_time": "2022-06-02 14:00:00",
+        "movie": None,
+        "cinema_hall": cinema_hall,
+    }
+    defaults.update(params)
+    return MovieSession.objects.create(**defaults)
+
+
+def image_upload_url(movie_id):
+    """Return URL for image upload"""
+    return reverse("cinema:movie-upload-image", args=[movie_id])
+
+
+def detail_url(movie_id):
+    return reverse("cinema:movie-detail", args=[movie_id])
+
+
+class UnauthenticatedMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AuthenticatedMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test_user1@test.com",
+            password="test_password1",
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_movie_list(self):
+        sample_movie()
+        movie_with_genres = sample_movie()
+        movie_with_actors = sample_movie()
+
+        genre = sample_genre()
+        actor = sample_actor()
+
+        movie_with_genres.genres.add(genre)
+        movie_with_actors.actors.add(actor)
+
+        response = self.client.get(MOVIE_URL)
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+    def test_filter_movies_by_title(self):
+        movie_1 = sample_movie(title="Fury")
+        movie_2 = sample_movie(title="Fast & Furious")
+
+        response = self.client.get(MOVIE_URL, {"title": "Fur"})
+
+        serializer_1 = MovieListSerializer(movie_1)
+        serializer_2 = MovieListSerializer(movie_2)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(serializer_1.data, response.data)
+        self.assertIn(serializer_2.data, response.data)

```

Comment:
The test `test_filter_movies_by_title` seems to be incorrect. The current logic checks if both `movie_1` and `movie_2` are in the response data, but the filter is for titles containing 'Fur'. This would mean only `movie_1` should be in the response, not `movie_2`. Please verify the filtering logic.